src/Carbon/Traits/Units.php

```diff
@@ -353,7 +353,7 @@ public function addUnit(Unit|string $unit, $value = 1, ?bool $overflow = null):
         $previousException = null;
 
         try {
-            $date = self::rawAddUnit($date, $unit, $value);
+            $date = self::rawAddUnit($date, $unit, interval($value));
```

Comment:
`interval()` does not exist, I guess you mean `intval`, but it needs to support float at this point:
```suggestion
            $date = self::rawAddUnit($date, $unit, (float) $value);
```

Reply:
Yes, sorry my bad. I meant `intval($value)`. Autofill syntax has changed to interval($value).
Got it, so could you please add `(float) $value` .
I'm using Carbon in my project where I'm getting a value from the config file / .env, though it is a interger value somehow it is treated as string and the carbon library is failing. So adding `intval($value)` worked in my project, however as you said it needs to support float as well. So making it to `(float) $value` as you said is best I think.